### PR TITLE
fix: NavTree fixes & increments

### LINKED ISSUE #7421: [navtree] keyboard navigation
- [x] introduce arrow navigation to navtree so that it can be much more quickly traversed from the keyboard without stepping through all menus
- [x] up and down arrows should traverse tree items in order
- [x] left and right arrows?
  - [x] expand/collapse on branch items?

### LINKED ISSUE #7422: [navtree] smarter opening of branches
- [x] remember previously open branches
- [x] open branches when creating a new object
- [ ] #7481

### LINKED ISSUE #7425: [deck] invoking the heading level menu in a stack section removes the attention treatment
- menu needs an attendable id
- also consider:
  - attention when clicking on plank heading
  - attention is currently lost when re-sizing planks

### LINKED ISSUE #7441: [navtree] presence is not working
even when there's presence in the plank heading the navtree does not reflect that

Image

### PR DESCRIPTION
**Pull request by thure**
This PR:
- resolves #7441
- resolves #7425 
- resolves #7422
- resolves #7421

https://github.com/user-attachments/assets/b71946c3-5831-4c12-acc5-7fa489a1264d

https://github.com/user-attachments/assets/674fd2b2-251b-4f89-96a5-a574dee497b9

Screenshot 2024-08-15 at 17 59 45

### CODE REVIEW

File: packages/apps/plugins/plugin-deck/src/DeckPlugin.tsx [hunk 1]
```diff
@@ -323,6 +324,14 @@ export const DeckPlugin = ({
             }
 
             case NavigationAction.OPEN: {
+              if (intent.data?.object) {
+                // forward to `EXPOSE`
+                void intentPlugin?.provides.intent.dispatch({
+                  action: NavigationAction.EXPOSE,
+                  data: { id: fullyQualifiedId(intent.data.object) },
+                });
+              }
```
Comment (ZaymonFC) [inline packages/apps/plugins/plugin-deck/src/DeckPlugin.tsx:333]: Add to list of returned intents instead?

**Review by ZaymonFC — APPROVED** [review]
LGTM, just one suggestion.

File: packages/apps/plugins/plugin-deck/src/DeckPlugin.tsx (hunk 1 above)
Comment (thure) [reply packages/apps/plugins/plugin-deck/src/DeckPlugin.tsx:333]: Ah smart, yes.

File: packages/sdk/app-framework/src/plugins/common/navigation.ts [hunk 2]
```diff
@@ -156,6 +156,7 @@ export enum NavigationAction {
   SET = `${NAVIGATION_ACTION}/set`,
   ADJUST = `${NAVIGATION_ACTION}/adjust`,
   CLOSE = `${NAVIGATION_ACTION}/close`,
+  EXPOSE = `${NAVIGATION_ACTION}/expose`,
```
Comment (wittjosiah) [inline packages/sdk/app-framework/src/plugins/common/navigation.ts:159]: I'm skeptical about this being a navigation action, but we already have a bunch of navigation actions that are pretty specific to deck so not too worried about it at this stage

**Pull request merged**